File: src/main/java/j/chain_of_resposibility/like/functions/LimitSupport.java
package j.chain_of_resposibility.like.functions;

public final class LimitSupport {
  public static Support create(String name, int limit) {
    return trouble -> {
      if (trouble.getNumber() < limit) {
        MessagePrinter.done(name, trouble);
        return ControlBreak.of(trouble).withControlBreak(true);
      } else {
        return ControlBreak.of(trouble);
      }
    };
  }
}

File: src/main/scala/s/chain_of_responsibility/like/functional/LimitSupport.scala
package s.chain_of_responsibility.like.functional

import s.chain_of_responsibility.Trouble

object LimitSupport {
  def create(name: String, limit: Int): PartialFunction[Trouble, Unit] = {
    case trouble if trouble.number < limit =>
      MessagePrinter.done(name, trouble)
  }
}
